### PR TITLE
Fix V2 Parquet page alignment for use with zStandard compression

### LINKED ISSUE #14781: [BUG] Parquet V2 page headers cannot currently be used with zStandard compression
**Describe the bug**
When using V2 page headers, the repetition and definition level data is not compressed. The Parquet writer currently increments the data pointer past the level data before sending to the compression function. The zStandard compression, however, requires input to be aligned on a 4 byte boundary, so simply incrementing the pointer past the level data often results in misaligned access errors.

**Additional context**
Current workaround PR #14772

### PR DESCRIPTION
**Pull request by etseidl**
## Description
Fixes #14781 

This PR makes changes to the Parquet writer to ensure that data to be compressed is properly aligned. Changes have also been made to the `EncPage` struct to make it easier to keep fields in that struct aligned, and also to reduce confusing re-use of fields. In particular, the `max_data_size` field can be any of a) the maximum possible size for the page data, b) the actual size of page data after encoding, c) the actual size of compressed page data. The latter two now have their own fields, `data_size` and `comp_data_size`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by hyperbolic2346 — APPROVED** [review]

File: cpp/src/io/parquet/page_enc.cu [hunk 1]
```diff
@@ -2708,19 +2736,26 @@ CUDF_KERNEL void __launch_bounds__(1024)
...
     // Copy page header
     hdr_len = page_g.hdr_size;
     memcpy_block<1024, true>(dst, src, hdr_len, t);
     src += page_g.max_hdr_size;
     dst += hdr_len;
-    // Copy page data
     uncompressed_size += hdr_len;
-    data_len = page_g.max_data_size;
+    data_len = ck_g.is_compressed ? page_g.comp_data_size : page_g.data_size;
+    // Copy page data. For V2, the level data and page data are disjoint.
```
Comment (vuule) [inline cpp/src/io/parquet/page_enc.cu:2747]: disjoint because of the alignment padding?

Comment (etseidl) [reply cpp/src/io/parquet/page_enc.cu:2747]: Yes. I suppose there will be cases where they are actually contiguous, but I don't know how worth it it is to test for that.

**Review by vuule — APPROVED** [review]

**Comment by vuule** [comment]
trying to check if the bot works with the command at the end of the comment
/ok to test

**Pull request merged**